### DISCUSSION TITLE
Programmatically creating the vue-leaflet map

### DISCUSSION THREAD

**Discussion by smithGit** [vue-leaflet/vue-leaflet#182] [Q&A]
There is a very good example app for vue-leaflet in which the map is created declaratively in the template:
https://github.com/vue-leaflet/vue3-demo-project/blob/master/src/App.vue 
However, are there any examples where the map is created programmatically in JavaScript?  I cannot figure out how to do that.  
Thanks to all.  
1/12/22: More specifically, here is what I think I need, trying to use the D3 graphics system with vue-leaflet:  The ability to use getPanes:
    var map = L.map('map')   <<< How is this done in vue-leaflet?  (If the map is created in the template, how can it be accessed?)
        .addLayer(osmTiles)
        .setView([ myGeoJson[idxjg].ctrLat, myGeoJson[idxjg].ctrLng], myGeoJson[idxjg].zoom)
        // .setView([40.72332345541449, -73.99], 14);

    // we will be appending the SVG to the Leaflet map pane
    // g (group) element will be inside the svg 
    var svg = d3.select(map.getPanes().overlayPane).append("svg");  << This is the key command I need to overlay svg on map

**Comment by CorzaHorizon** [comment]
```
<l-map ref="map" @ready="onMapReady"></l-map>

// script

setup() {
    const map = ref(null);
    const onMapReady = async () => {
      console.log(map);
    };
    return { map, onMapReady };
},
```
or 

```
<div ref="map_div"></div>

//script

setup() {
  const map_div = ref(null);
  return { map_div };
},
mounted() {
  this.map_div.id = 'map';
  const map = L.map('map').setView([51.505, -0.09], 13);

  L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
      attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
  }).addTo(map);
  
  L.marker([51.5, -0.09]).addTo(map)
      .bindPopup('A pretty CSS3 popup.<br> Easily customizable.')
      .openPopup();
}

```
